src/Makefile.gtest.include

```diff
@@ -10,6 +10,7 @@ zen_gtest_SOURCES = \
 	gtest/tx_creation_utils.cpp \
 	gtest/test_checktransaction.cpp \
 	gtest/test_consensus.cpp \
+	gtest/test_createnewblock.cpp \
```

Comment:
Please move this test to the bottom as reported below:

```
...
	gtest/test_asyncproofverifier.cpp \
	gtest/test_blockdownload.cpp \
	gtest/test_createnewblock.cpp
...
```

In this way we can simplify the PR so that the following modifications can be reverted (not needed anymore):
- `src/gtest/test_mempool.cpp`
- `src/util.cpp/src/util.h`
- `src/wallet/gtest/test_deadlock.cpp`
- `src/wallet/gtest/test_wallet_zkeys.cpp`

I have tested multiple times on my machine running full test suite as well as single tests and no error shows; also the CI is fine.

As a final comment, for sure we can consider making the test suite more stable and reliable; #1539 (https://github.com/zcash/zcash/issues/1539) from ZCash repo for reference.

Reply:
Done

Reply:
I second @JackPiri's suggestion, but I cannot see it implemented in the code. Could you please double check?